### PR TITLE
[IE CLDNN] Batched blob support is added for NV12

### PR DESCRIPTION
**Pull request by isanghao**
### Details:
- Batched nv12 blob support from GPU
 
### Tickets:
 - 51590

### CODE REVIEW

File: inference-engine/src/cldnn_engine/cldnn_infer_request.cpp [hunk 1]
```diff
@@ -322,6 +323,20 @@ void CLDNNInferRequest::copyInputData(std::shared_ptr<cldnn::network> network,
     }
 }
 
+void checkInputBlobNV12(const NV12Blob *nv12_ptr) {
+    auto y_ptr = nv12_ptr->y()->as<gpu::ClBlob>();
+
+    // if the blobs are not remote, check their size
+    if (!y_ptr) {
+        if (nv12_ptr->y()->buffer() == nullptr) IE_THROW() << str_not_allocated;
```
Comment (ilya-lavrenov) [inline inference-engine/src/cldnn_engine/cldnn_infer_request.cpp:331]: -> `IE_THROW(NotAllocated) << str_not_allocated;`

Common recommendation - use specific exceptions if possible, while `IE_THROW()` throws a generic one..

**Comment by vladimir-paramuzov** [comment]
@isanghao  please pay attention that one GPU test failed: smoke_SetBlobOfKindGPU/SetBlobOfKindTest.CompareWithRefs/Kind=BatchOfSimple

File: inference-engine/src/cldnn_engine/cldnn_infer_request.cpp (hunk 1 above)
Comment (isanghao) [reply inference-engine/src/cldnn_engine/cldnn_infer_request.cpp]: Applied, thanks!

File: inference-engine/src/cldnn_engine/cldnn_infer_request.cpp [hunk 2]
```diff
@@ -336,21 +351,21 @@ void checkInputBlob(const Blob::Ptr &blob,
...
-        // if the blobs are not remote, check their size
-        if (!y_ptr) {
-            if (nv12_ptr->y()->buffer() == nullptr) IE_THROW() << str_not_allocated;
+        if (nv12_ptr != nullptr) {
+            checkInputBlobNV12(nv12_ptr);
+        } else if (batched_ptr != nullptr) {
+            for (auto i = 0; i < batched_ptr->size(); i++) {
+                nv12_ptr = batched_ptr->getBlob(i)->as<NV12Blob>();
+                if (nv12_ptr == nullptr)
+                    IE_THROW() << wrong_batched_blob;
```
[the remark this answers is not in the document]
Comment (isanghao) [reply inference-engine/src/cldnn_engine/cldnn_infer_request.cpp]: Changed to `unsupported_batched_blob`. Thanks.

File: inference-engine/src/cldnn_engine/cldnn_infer_request.cpp [hunk 3]
```diff
@@ -336,21 +351,21 @@ void checkInputBlob(const Blob::Ptr &blob,
...
-            IE_THROW(ParameterMismatch) << wrong_nv12_blob;
-        }
-
-        auto y_ptr = nv12_ptr->y()->as<gpu::ClBlob>();
+        auto batched_ptr = blob->as<BatchedBlob>();
 
-        // if the blobs are not remote, check their size
-        if (!y_ptr) {
-            if (nv12_ptr->y()->buffer() == nullptr) IE_THROW() << str_not_allocated;
+        if (nv12_ptr != nullptr) {
```
[the remark this answers is not in the document]
Comment (isanghao) [reply inference-engine/src/cldnn_engine/cldnn_infer_request.cpp]: applied, thanks.

**Comment by isanghao** [comment]
> @isanghao please pay attention that one GPU test failed: smoke_SetBlobOfKindGPU/SetBlobOfKindTest.CompareWithRefs/Kind=BatchOfSimple

As discussed, `BatchOfSimple` is not supported currently. So I disabled it for GPU.

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**